Formula/kfctl.rb

```diff
@@ -0,0 +1,20 @@
+class Kfctl < Formula
+  desc "Kubeflow CLI tool"
+  homepage "https://www.kubeflow.org"
+  url "https://github.com/kubeflow/kubeflow/releases/download/v0.5.1/kfctl_v0.5.1_darwin.tar.gz"
+  version "0.5.1"
+  sha256 "37790a2bd02a27f94e97c550bc6743a8bdbf69751e78d9e968d018c3bf2bc503"
+  head "https://github.com/kubeflow/kubeflow.git"
+
+  bottle :unneeded
+
+  depends_on "kubernetes-cli"
+
+  def install
+    bin.install "kfctl"
+  end
+
+  test do
+    assert_match "v20181207", shell_output("#{bin}/kfctl version 2>&1")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Makes sense. I have to admit that I mostly based all of it on the [kubectx formula](https://github.com/Homebrew/homebrew-core/blob/master/Formula/kubectx.rb). Might have picked a particularly bad example.

Reply:
Older formula might have worse tests (something is better then nothing). But for new additions we require some tested functions

Reply:
there is a [Makefile](https://github.com/kubeflow/kubeflow/blob/master/bootstrap/Makefile) that builds [kfctl](https://github.com/kubeflow/kubeflow/blob/master/bootstrap/Makefile#L90). Does this need to be a bottle?